addons/mail/tests/test_res_partner.py

```diff
@@ -90,6 +90,33 @@ def test_parse_partner_name(self):
             self.assertEqual(parsed[0], expected_name)
             self.assertEqual(parsed[1], expected_email)
 
+
+    def test_res_partner_address_tracking(self):
+        company_partner = self.env.company.partner_id
+        # use some wacky formatting to check inlining
+        company_partner.country_id.address_format = """%(street)s
+            \n\n\n%(street2)s
+            %(city)s %(state_code)s %(zip)s
+            \n%(country_name)s\n
+            \n   """
+        company_partner.type = 'contact'
+        company_partner.street = 'Some Street Name'
+        partner = self.env['res.partner'].create({
+            'name': 'Some Guy',
+            'parent_id': company_partner.id,
+        })
+        self.env.flush_all()
+        self.cr.precommit.run()
+
+        company_partner.street = 'Some Other Street Name'
```

Comment:
Would change another field than street at the same time to make sure we don't generate multiple tracking messages ("city" for example).

Would also be interesting to count tracking messages generated, in the same idea, rather than just checking that it's "inside the changes".